src/zarr/core/metadata/v3.py

```diff
@@ -215,6 +243,9 @@ def from_dict(cls, data: dict[str, JSON]) -> Self:
         # check that the node_type attribute is correct
         _ = parse_node_type_array(_data.pop("node_type"))
 
+        # check that the data_type attribute is valid
+        _ = DataType(_data["data_type"])
```

Comment:
why not use `parse_dtype` here?

Reply:
alternatively, why don't we rely on the invocation of `parse_dtype` that will occur when we call `cls(**_data)`

Reply:
Good question.

`parse_dtype` is used to parse data coming from either user code (e.g. `f8`) or from array metadata (where the value must be `float64` for example). `from_dict` is the one place we can intercept data coming from JSON.

Reply:
Without this like, we would silently handle a dtype like `f8` coming from 
`zarr.json`.

Reply:
I see, so we need separate "parse this dtype-like thing that a user provided" logic from "validate the `dtype` field in this zarr.json document" logic. I feel like the best way to do this is with a function called `validate_dtype` or something that ensures that the input is from the finite set of dtype symbols allowed in the spec. but I think that's a project for another PR, and doesn't block this one